src/main/java/com/rybka/Application.java

```diff
@@ -1,75 +1,37 @@
 package com.rybka;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.dataformat.csv.CsvMapper;
-import com.rybka.command.ConvertCommand;
-import com.rybka.command.ExportCommand;
-import com.rybka.command.HistoryCommand;
+import com.rybka.command.Command;
 import com.rybka.config.*;
-import com.rybka.dao.CurrencyHistoryDAO;
-import com.rybka.exception.InvalidPropertyException;
-import com.rybka.service.connector.ExchangeRateConnector;
-import com.rybka.service.exchange.ExchangeService;
-import com.rybka.service.connector.PrimeExchangeRateConnector;
-import com.rybka.service.export.CSVExportService;
-import com.rybka.service.export.ConsoleExportService;
-import com.rybka.service.export.JSONExportService;
+import com.rybka.service.connector.BaseCurrencyExchangeConnector;
+import com.rybka.service.export.ExportService;
 import com.rybka.view.ExchangeView;
-import coresearch.cvurl.io.request.CVurl;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 
-import java.net.http.HttpClient;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.Map;
-import java.util.Scanner;
 
 public class Application {
 
     public static void main(String[] args) {
 
-        var objectMapper = new ObjectMapper();
+        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(ApplicationConfiguration.class);
 
-        var primeExchangeRateConnector = new PrimeExchangeRateConnector(HttpClient.newHttpClient(), objectMapper);
-        var exchangeRateConnectorConnector = new ExchangeRateConnector(new CVurl());
+        Map<String, BaseCurrencyExchangeConnector> exchangeSourceMap = Map.of(
```

Comment:
Try to get rid off the map by using ConditionalOnProperty

https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnProperty.html

Reply:
fixed

Reply:
ok